src/lazy_buffer.rs

```diff
@@ -44,6 +43,25 @@ where
             }
         }
     }
+
+    pub fn prefill(&mut self, len: usize) -> bool {
+        if self.done {
+            return self.buffer.len() >= len;
+        }
+
+        if len > self.buffer.len() {
+            let extra = len - self.buffer.len();
+            for _ in 0..extra {
```

Comment:
Could we avoid `extra` and instead iterate over `self.buffer.len() .. len`?

Reply:
yes, missed that!